### PR TITLE
ISSUES-693 add SIPHash function

### LINKED ISSUE #693: Support SIPHASH function
**Summary**
```text
mysql> SELECT SIPHASH('1234567890');
+----------------------+
| SIPHASH(1234567890)  |
+----------------------+
| 18110648197875983073 |
+----------------------+
1 row in set (0.00 sec)

mysql> SELECT SIPHASH(1);
+---------------------+
| SIPHASH(1)          |
+---------------------+
| 2206609067086327257 |
+---------------------+
1 row in set (0.01 sec)

mysql> SELECT SIPHASH(1.2);
+---------------------+
| SIPHASH(1.2)        |
+---------------------+
| 2854037594257667269 |
+---------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by zhang2014**
## Summary

 Add SIPHash function

## Changelog

- New Feature

## Related Issues

Fixes #693

### CODE REVIEW

**Review by BohuTANG — APPROVED** [review]
LGTM

File: common/functions/src/hashes/siphash.rs [hunk 1]
```diff
@@ -40,13 +48,31 @@ impl IFunction for SipHashFunction {
         }
 
         match args[0] {
+            DataType::Int8 => Ok(DataType::UInt64),
+            DataType::Int16 => Ok(DataType::UInt64),
```
Comment (sundy-li) [inline common/functions/src/hashes/siphash.rs:52]: DataType::Int8 | DataType::Int16 ...

File: common/functions/src/hashes/siphash.rs [hunk 2]
```diff
@@ -32,6 +36,10 @@ impl IFunction for SipHashFunction {
         "siphash"
     }
 
+    fn num_arguments(&self) -> usize {
+        1
+    }
+
     fn return_type(&self, args: &[DataType]) -> Result<DataType> {
         if args.len() != 1 {
```
Comment (sundy-li) [inline common/functions/src/hashes/siphash.rs:44]: it's already checked in `num_arguments`, no need to check again

**Pull request merged**